### ISSUE TITLE
The `add_fields` transform should add metric "fields"

### ISSUE THREAD

**Issue by binarylogic** [timberio/vector#524]
This is blocked by https://github.com/timberio/vector/issues/512

The `add_fields` transform should add fields to the metric data type as well.

**Comment by binarylogic** [comment]
I'm starting to think this does not make sense, because the `add_fields` transform supports nested fields, arrays, and other concepts that do not make sense for metrics tags. It might be better to create a separate `add_tags` transform.

**Comment by lukesteensen** [comment]
Agreed on the points above. Closing this in favor of #771.
